### PR TITLE
feat: ability to point header logo to other locations

### PR DESCRIPTION
**Pull request by arora-r**
### Description

Previously, the top left logo in the LMS header was hard coded to redirect learners to the `/dashboard` page. This PR introduces a configurable option (`LOGO_DESTINATION`) allowing administrators to customize the destination URL for the logo link. Now, learners can be redirected to any specified custom link, providing greater flexibility and customization within the LMS.

### Code changes

- Added a new setting (`LOGO_DESTINATION`) to dynamically control the destination URL of the logo in the header.
- By utilizing an environment variable, we've simplified the process of configuring the logo link.
- The default behavior (when the environment variable is not specified) remains unchanged, directing users to the LMS dashboard to maintain continuity with existing workflows.

### CODE REVIEW

**Comment by openedx-webhooks** [comment]
Thanks for the pull request, @arora-r! Please note that it may take us up to several weeks or months to complete a review and merge your PR.

Feel free to add as much of the following information to the ticket as you can:
- supporting documentation
- [Open edX discussion forum threads](https://discuss.openedx.org/)
- timeline information ("this must be merged by XX date", and why that is)
- partner information ("this is a course on edx.org")
- any other information that can help Product understand the context for the PR

All technical communication about the code itself will be done via the GitHub pull request interface. As a reminder, [our process documentation is here](http://edx-developer-guide.readthedocs.org/en/latest/process/overview.html).


Please let us know once your PR is ready for our review and all tests are green.


:warning: We can't start reviewing your pull request until you've submitted a     [signed contributor agreement](https://openedx.org/cla)     or indicated your institutional affiliation. Please see the [CONTRIBUTING](https://github.com/openedx/.github/blob/master/CONTRIBUTING.md)     file for more information.     If you've signed an agreement in the past, you may need to re-sign.  See     [The New Home of the Open edX Codebase](https://open.edx.org/blog/the-new-home-of-the-open-edx-codebase/)     for details.

    
Once you've signed the CLA, please allow 1 business day for it to be processed.  After this time, you can re-run the     CLA check by adding a comment here that you have signed it.  If the problem persists, you can tag the `@openedx/cla-problems` team in a     comment on your PR for further assistance.

**Comment by mphilbrick211** [comment]
Hi @arora-r! Thank you for this contribution! Please let me know if you have any questions regarding submitting a [CLA form](https://openedx.atlassian.net/wiki/spaces/COMM/pages/941457737/How+to+start+contributing+to+the+Open+edX+code+base).

**Comment by arora-r** [comment]
Hey @mphilbrick211, any update on this issue - would we be able to get someone to review this and approved?

**Comment by mphilbrick211** [comment]
Hi @arora-r! Thanks for your patience. We're looking into getting a reviewer for you. Thanks!

**Comment by arora-r** [comment]
Hey @mphilbrick211 any update?

**Review by arbrandes — CHANGES_REQUESTED** [review]
I find the underlying premise is sound: we probably should allow people to change the logo URL.

At the same time, we find ourselves in a quagmire of toggles and configuration - some of it in the backend, some in the frontend - that we're actively trying to counter by the introduction of the (much) more flexible approach of plugin slots.  You can find more details about how we're handling it for the footer here: https://github.com/openedx/wg-frontend/issues/178.

In other words, would you consider your issue to be solved if you could create a plugin to override whatever is in the header as needed?

**Comment by arbrandes** [comment]
There are very similar conversations happening around the footer:

* https://github.com/openedx/frontend-component-footer/pull/405
* https://github.com/openedx/frontend-component-footer/pull/403
* https://github.com/openedx/platform-roadmap/issues/324

It can be gleaned from my comments there (and here) what my general opinion is: we should spring for plugins sooner rather than later in order to avoid adding yet more granular configuration.  That doesn't mean my opinion is the only one that counts, though.  Plus, the header is more complicated than the footer and will need more careful thought as to what sort of props we'll need to pass down into potential plugins.

I guess what I'm trying to say is that I'd like more input from the community before making a decision as maintainer.  I'm going to start a thread in the forum and post the link here.

**Comment by arbrandes** [comment]
As promised, here's the thread where we can continue the general conversation:

https://discuss.openedx.org/t/plugin-slots-vs-configuration/13009

**Comment by mphilbrick211** [comment]
@arora-r @arbrandes - is this still in progress, or can it be closed? Thanks!

**Comment by mphilbrick211** [comment]
Closing due to inactivity.
